### PR TITLE
Add methods to the account, project and server clients

### PR DESCRIPTION
**Pull request by acroz**
These additions facilitate updating some internal tooling, in particular removing a dependency on a legacy codebase.

### CODE REVIEW

**Review by imrehg — APPROVED** [review]
LGTM, though not sure I'm qualified to comment on the correctness of the API calls themselves (the fields that are returned)

One question, the new `list_all` call has this note:

> This method requires administrative privileges not available to most users.

Would it be needed & possible to test the call with an unprivileged user as well? Or since the tests here are mocked, testing that is out of scope?

**Comment by acroz** [comment]
@imrehg I've tested the integration against the actual backend services myself. You're right though that the automated tests in this repo mock out the actual call to the backend and so nothing special is required in this case.

File: faculty/clients/project.py [hunk 1]
```diff
@@ -16,15 +16,34 @@
...
+
+    Parameters
+    ----------
+    id : uuid.UUID
+        The ID of the project.
+    name : str
+        The name of the project.
+    owner_id : uuid.UUID
+        The user ID of the owner of the project.
+    archived_at : datetime, optional
```
Comment (tomas-milata) [inline faculty/clients/project.py:38]: Should this be `datetime.datetime` given that we use the full name for e.g. `uuid.UUID`?

File: faculty/clients/project.py [hunk 2]
```diff
@@ -115,12 +134,32 @@ def list_accessible_by_user(self, user_id):
...
+        include_archived : bool, optional
+            If True, return archived projects. Default: False.
+
+        Returns
+        -------
+        List[Project]
+            The projects in Faculty.
+        """
+        params = {"includeArchived": int(include_archived)}
+        return self._get("/project", _ProjectSchema(many=True), params=params)
```
Comment (tomas-milata) [inline faculty/clients/project.py:154]: AFAIK, the backend accepts something like `GET /project?includeArchived=true`. Is this the resulting URL? Is there something about some bool/int conversion I'm missing?

**Review by tomas-milata — COMMENTED** [review]
Had some questions but LGTM otherwise!

File: faculty/clients/project.py (hunk 2 above)
Comment (acroz) [reply faculty/clients/project.py:154]: The backend accepts "true", "false", "1" or "0", so casting the bools to ints seemed most pragmatic.

**Comment by tomas-milata** [comment]
Oh, I didn't know Play worked that way. 👍
<https://emojipedia.org/thumbs-up/>

File: faculty/clients/project.py (hunk 2 above)
Comment (tomas-milata) [reply faculty/clients/project.py:154]: Oh, I didn't know Play worked that way. 👍

**Pull request merged**